internal/provider/resource_hvn.go

```diff
@@ -119,18 +122,29 @@ func resourceHvnCreate(ctx context.Context, d *schema.ResourceData, meta interfa
 
 	hvnID := d.Get("hvn_id").(string)
 	cidrBlock := d.Get("cidr_block").(string)
+	projectID := d.Get("project_id").(string)
+	resourceProjectID := d.Get("project_id").(string)
 
+	// TODO: would we want to conditionally set project id if defined in resource with d.get and otherwise grab project id
+	// with client.config.projectid
 	loc := &sharedmodels.HashicorpCloudLocationLocation{
 		OrganizationID: client.Config.OrganizationID,
-		ProjectID:      client.Config.ProjectID,
 		Region: &sharedmodels.HashicorpCloudLocationRegion{
 			Provider: d.Get("cloud_provider").(string),
 			Region:   d.Get("region").(string),
 		},
 	}
 
+	projectID, err := GetProjectID(resourceProjectID, client.Config.ProjectID)
+	if err != nil {
+		return diag.Errorf("unable to retrieve project ID: %v", err)
+	}
+
+	loc.ProjectID = projectID
```

Comment:
Generally in golang, you want to declare all the variables before you initialize an object that contains them. So I would move L138-142 above the declaration of `loc`, and then set `ProjectID: projectID` in the object declaration.